### ISSUE TITLE
Client does not seem to work with IBM's api.ng.bluemix.net endpoint, which is API version 2.44.0

### ISSUE THREAD

**Issue by lastras** [cloudfoundry/cf-java-client#513]
I am trying to use this java client sdk to programmatically (say) create a service but I don't have a choice of CF API - I need to interface with an existing cf installation (api.ng.bluemix.net, IBM's bluemix). To verify the version that IBM is running, when I authenticate using the cf command line tool I get

```
API endpoint:   https://api.ng.bluemix.net (API version: 2.44.0)  
```

When I run this code against pivotal's endpoint (api.run.pivotal.io)

```
CloudFoundryClient cfc = SpringCloudFoundryClient.builder()
   .host("host")
   .username(“username")
   .password(“password")
   .build();
CloudFoundryOperations cfo = DefaultCloudFoundryOperations.builder()
   .cloudFoundryClient(cfc)
   .organization(“org")
   .space(“space")
   .build();

cfo.organizations().list().map( o -> o.getName() ).subscribe(System.err::println);;

```

I do get my organization listed. But when I run it against IBM's endpoint (api.ng.bluemix.net) there is no printout. I am including the 2.0.0 SNAPSHOT versions of the client & operations, as well as the 2.5.0 versions of reactor as recommended by README.md. 

I have verified that changing the host to something incorrect results in an error, but changing the username/password to incorrect values result in exactly the same silent (no printout) behavior. Thus I did a

```
System.out.println(cfo.SUPPORTED_CLI_VERSION);
```

and obtained 2.54.0 as the version.

This points out that there is a discrepancy in the APIs. In order to investigate further, I downloaded the latest version of the cf cli

```
./cf --version
cf version 6.19.0+b29b4e0-2016-06-08
```

and this cf did not have any difficulties authenticating with IBM's bluemix endpoint and listing services, etc., which makes me suspect that either the cf is more advanced at detecting older API versions (and managing that) or the CF APIs from 2.44 -> 2.54 are still compatible for what I am trying to do and instead there is a bug in the java client library (hence this issue).

**Comment by twoseat** [comment]
Hi @lastras - Thanks for the comprehensive write-up! I've done some initial testing and I've confirmed that your sample app returns values correctly when run against pcf 2.44. That leaves two alternatives:

1. IBM are running some modification that we're not allowing for (seems unlikely to me, but possible)

2. You're not actually making a connection. More promising...

It turns out that run in a simple setup like your sample we don't communicate a failed login in a useful way (caveat: I'm not familiar with this area of our code, so I need to do some more digging to work out what the 'correct' behaviour should be). If I connect with a bad username to 2.44 locally, or on bluemix, I see the same 'no printout' you mention.

Can you turn on debug logging* and post the trace here, which should show us whether you're making a connection? Thanks!

----------

*I'm running your sample as a Spring Boot app, so dropping the following into `/src/main/java/resources/logback.xml` worked for me:

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>

    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <pattern>%-23thread %-35logger %msg%n</pattern>
        </encoder>
    </appender>

    <logger name="reactor.io.netty" level="DEBUG"/>
    <logger name="stream" level="DEBUG"/>

    <root level="WARN">
        <appender-ref ref="STDOUT"/>
    </root>

</configuration>
```

**Comment by lastras** [comment]
Here it goes.... I can't say I understand much of what is going on.

```22:55:52.924 [main] DEBUG o.s.web.client.RestTemplate - Created GET request for "https://api.ng.bluemix.net:443/v2/info"
22:55:52.972 [main] DEBUG o.s.web.client.RestTemplate - Setting request Accept header to [application/json, application/*+json]
22:56:55.648 [main] DEBUG o.s.web.client.RestTemplate - GET request for "https://api.ng.bluemix.net:443/v2/info" resulted in 200 (OK)
22:56:55.653 [main] DEBUG o.s.web.client.RestTemplate - Reading [interface java.util.Map] as "application/json;charset=utf-8" using [org.springframework.http.converter.json.MappingJackson2HttpMessageConverter@13c9d689]
22:56:55.912 [main] DEBUG o.c.s.u.n.OAuth2RestTemplateBuilder - Modifying ObjectMapper configuration
22:56:56.076 [main] DEBUG r.core.util.Logger$LoggerFactory - Using Slf4j logging framework
22:56:56.256 [main] DEBUG i.n.u.i.l.InternalLoggerFactory - Using SLF4J as the default logging framework
22:56:56.272 [main] DEBUG io.netty.handler.ssl.OpenSsl - netty-tcnative not in the classpath; OpenSslEngine will be unavailable.
22:56:56.312 [main] DEBUG io.netty.handler.ssl.JdkSslContext - Default protocols (JDK): [TLSv1.2, TLSv1.1, TLSv1] 
22:56:56.312 [main] DEBUG io.netty.handler.ssl.JdkSslContext - Default cipher suites (JDK): [TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA, SSL_RSA_WITH_3DES_EDE_CBC_SHA]
22:56:56.320 [main] DEBUG reactor.io.netty.tcp.TcpClient - Connecting with SSL enabled using context JdkSslClientContext
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent0 - java.nio.Buffer.address: available
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: available
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent0 - sun.misc.Unsafe.copyMemory: available
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: true
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent0 - java.nio.DirectByteBuffer.<init>(long, int): available
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent - Java version: 8
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.noUnsafe: false
22:56:56.336 [main] DEBUG i.n.util.internal.PlatformDependent - sun.misc.Unsafe: available
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.noJavassist: false
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - Javassist: unavailable
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - You don't have Javassist in your class path or you don't have enough permission to load dynamically generated classes.  Please check the configuration for better performance.
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.bitMode: 64 (sun.arch.data.model)
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.noPreferDirect: false
22:56:56.340 [main] DEBUG i.n.util.internal.PlatformDependent - io.netty.maxDirectMemory: 14074511360 bytes
22:56:56.346 [main] DEBUG i.n.u.internal.NativeLibraryLoader - -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
22:56:56.346 [main] DEBUG i.n.u.internal.NativeLibraryLoader - -Dio.netty.native.workdir: /tmp (io.netty.tmpdir)
22:56:56.376 [main] DEBUG io.netty.util.NetUtil - Loopback interface: lo (lo, 0:0:0:0:0:0:0:1%lo)
22:56:56.386 [main] DEBUG io.netty.util.NetUtil - /proc/sys/net/core/somaxconn: 128
22:56:56.417 [main] DEBUG r.io.netty.util.NettyNativeDetector - Default Netty Epoll support : true
22:56:56.440 [main] DEBUG i.n.c.MultithreadEventLoopGroup - -Dio.netty.eventLoopThreads: 32
22:56:56.492 [main] DEBUG io.netty.channel.nio.NioEventLoop - -Dio.netty.noKeySetOptimization: false
22:56:56.492 [main] DEBUG io.netty.channel.nio.NioEventLoop - -Dio.netty.selectorAutoRebuildThreshold: 512
22:56:57.100 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.numHeapArenas: 32
22:56:57.100 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.numDirectArenas: 32
22:56:57.100 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.pageSize: 8192
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxOrder: 11
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.chunkSize: 16777216
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.tinyCacheSize: 512
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.smallCacheSize: 256
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.normalCacheSize: 64
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.maxCachedBufferCapacity: 32768
22:56:57.101 [main] DEBUG i.n.buffer.PooledByteBufAllocator - -Dio.netty.allocator.cacheTrimInterval: 8192
22:56:57.160 [main] DEBUG io.netty.channel.DefaultChannelId - -Dio.netty.processId: 19374 (auto-detected)
22:56:57.166 [main] DEBUG io.netty.channel.DefaultChannelId - -Dio.netty.machineId: 3e:45:1a:ff:fe:48:40:fb (auto-detected)
22:56:57.168 [main] DEBUG i.n.util.internal.ThreadLocalRandom - -Dio.netty.initialSeedUniquifier: 0xc135561dc8d1d34b
22:56:57.186 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.level: simple
22:56:57.186 [main] DEBUG io.netty.util.ResourceLeakDetector - -Dio.netty.leakDetection.maxRecords: 4
22:56:57.240 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.allocator.type: pooled
22:56:57.240 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.threadLocalDirectBufferSize: 65536
22:56:57.240 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.maxThreadLocalCharBufferSize: 16384
22:56:57.605 [reactor-tcp-client-io-1] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.bytebuf.checkAccessible: true
22:56:57.661 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0] REGISTERED
22:56:57.698 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0] CONNECT: api.ng.bluemix.net/169.54.245.68:443
22:56:57.746 [reactor-tcp-client-io-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxCapacity: 262144
22:56:57.746 [reactor-tcp-client-io-1] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.linkCapacity: 16
22:56:57.811 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0, L:/9.47.225.81:55181 - R:api.ng.bluemix.net/169.54.245.68:443] ACTIVE
22:56:57.858 [reactor-tcp-client-io-1] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:57.888 [reactor-tcp-client-io-1] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:57.932 [reactor-tcp-client-io-1] DEBUG io.netty.handler.ssl.SslHandler - [id: 0x2c1930d0, L:/9.47.225.81:55181 - R:api.ng.bluemix.net/169.54.245.68:443] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
22:56:57.932 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0, L:/9.47.225.81:55181 - R:api.ng.bluemix.net/169.54.245.68:443] USER_EVENT: SslHandshakeCompletionEvent(SUCCESS)
22:56:57.959 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8] REGISTERED
22:56:57.984 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8] CONNECT: api.ng.bluemix.net/169.54.245.68:443
22:56:57.976 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd] REGISTERED
22:56:57.988 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd] CONNECT: api.ng.bluemix.net/169.54.245.68:443
22:56:57.992 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8, L:/9.47.225.81:55184 - R:api.ng.bluemix.net/169.54.245.68:443] ACTIVE
22:56:57.996 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd, L:/9.47.225.81:55185 - R:api.ng.bluemix.net/169.54.245.68:443] ACTIVE
Done!!
reached the end of the main!!!
22:56:58.072 [reactor-tcp-client-io-1] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Retrieving token from https://uaa.ng.bluemix.net/oauth/token
22:56:58.076 [reactor-tcp-client-io-2] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.076 [reactor-tcp-client-io-1] DEBUG o.s.web.client.RestTemplate - Created POST request for "https://uaa.ng.bluemix.net/oauth/token"
22:56:58.076 [reactor-tcp-client-io-1] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Encoding and sending form: {grant_type=[password], username=[lastrasl@us.ibm.com], password=[B0tsRage]}
22:56:58.082 [reactor-tcp-client-io-3] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.129 [reactor-tcp-client-io-2] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.130 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3] REGISTERED
22:56:58.130 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3] CONNECT: api.ng.bluemix.net/169.54.245.68:443
22:56:58.157 [reactor-tcp-client-io-3] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.166 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3, L:/9.47.225.81:55186 - R:api.ng.bluemix.net/169.54.245.68:443] ACTIVE
22:56:58.188 [reactor-tcp-client-io-2] DEBUG io.netty.handler.ssl.SslHandler - [id: 0xabd28cb8, L:/9.47.225.81:55184 - R:api.ng.bluemix.net/169.54.245.68:443] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
22:56:58.188 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8, L:/9.47.225.81:55184 - R:api.ng.bluemix.net/169.54.245.68:443] USER_EVENT: SslHandshakeCompletionEvent(SUCCESS)
22:56:58.193 [reactor-tcp-client-io-2] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Retrieving token from https://uaa.ng.bluemix.net/oauth/token
22:56:58.193 [reactor-tcp-client-io-2] DEBUG o.s.web.client.RestTemplate - Created POST request for "https://uaa.ng.bluemix.net/oauth/token"
22:56:58.193 [reactor-tcp-client-io-2] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Encoding and sending form: {grant_type=[password], username=[lastrasl@us.ibm.com], password=[B0tsRage]}
22:56:58.200 [reactor-tcp-client-io-3] DEBUG io.netty.handler.ssl.SslHandler - [id: 0x69fdf2dd, L:/9.47.225.81:55185 - R:api.ng.bluemix.net/169.54.245.68:443] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
22:56:58.200 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd, L:/9.47.225.81:55185 - R:api.ng.bluemix.net/169.54.245.68:443] USER_EVENT: SslHandshakeCompletionEvent(SUCCESS)
22:56:58.200 [reactor-tcp-client-io-3] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Retrieving token from https://uaa.ng.bluemix.net/oauth/token
22:56:58.200 [reactor-tcp-client-io-3] DEBUG o.s.web.client.RestTemplate - Created POST request for "https://uaa.ng.bluemix.net/oauth/token"
22:56:58.200 [reactor-tcp-client-io-3] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Encoding and sending form: {grant_type=[password], username=[lastrasl@us.ibm.com], password=[B0tsRage]}
22:56:58.220 [reactor-tcp-client-io-4] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.251 [reactor-tcp-client-io-4] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.295 [reactor-tcp-client-io-4] DEBUG io.netty.handler.ssl.SslHandler - [id: 0xd90b01c3, L:/9.47.225.81:55186 - R:api.ng.bluemix.net/169.54.245.68:443] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
22:56:58.295 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3, L:/9.47.225.81:55186 - R:api.ng.bluemix.net/169.54.245.68:443] USER_EVENT: SslHandshakeCompletionEvent(SUCCESS)
22:56:58.296 [reactor-tcp-client-io-4] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Retrieving token from https://uaa.ng.bluemix.net/oauth/token
22:56:58.298 [reactor-tcp-client-io-4] DEBUG o.s.web.client.RestTemplate - Created POST request for "https://uaa.ng.bluemix.net/oauth/token"
22:56:58.298 [reactor-tcp-client-io-4] DEBUG o.s.s.o.c.t.g.p.ResourceOwnerPasswordAccessTokenProvider - Encoding and sending form: {grant_type=[password], username=[lastrasl@us.ibm.com], password=[B0tsRage]}
22:56:58.514 [reactor-tcp-client-io-1] DEBUG o.s.web.client.RestTemplate - POST request for "https://uaa.ng.bluemix.net/oauth/token" resulted in 401 (Unauthorized); invoking error handler
22:56:58.567 [reactor-tcp-client-io-3] DEBUG o.s.web.client.RestTemplate - POST request for "https://uaa.ng.bluemix.net/oauth/token" resulted in 401 (Unauthorized); invoking error handler
22:56:58.593 [reactor-tcp-client-io-2] DEBUG o.s.web.client.RestTemplate - POST request for "https://uaa.ng.bluemix.net/oauth/token" resulted in 401 (Unauthorized); invoking error handler
22:56:58.636 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0, L:/9.47.225.81:55181 - R:api.ng.bluemix.net/169.54.245.68:443] CLOSE
22:56:58.642 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8, L:/9.47.225.81:55184 - R:api.ng.bluemix.net/169.54.245.68:443] CLOSE
22:56:58.664 [reactor-tcp-client-io-4] DEBUG o.s.web.client.RestTemplate - POST request for "https://uaa.ng.bluemix.net/oauth/token" resulted in 401 (Unauthorized); invoking error handler
22:56:58.685 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3, L:/9.47.225.81:55186 - R:api.ng.bluemix.net/169.54.245.68:443] CLOSE
22:56:58.688 [reactor-tcp-client-io-1] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.688 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0, L:/9.47.225.81:55181 ! R:api.ng.bluemix.net/169.54.245.68:443] INACTIVE
22:56:58.688 [reactor-tcp-client-io-1] DEBUG reactor.io.netty.http.HttpClient - [id: 0x2c1930d0, L:/9.47.225.81:55181 ! R:api.ng.bluemix.net/169.54.245.68:443] UNREGISTERED
22:56:58.689 [reactor-tcp-client-io-2] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.690 [reactor-tcp-client-io-4] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.690 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3, L:/9.47.225.81:55186 ! R:api.ng.bluemix.net/169.54.245.68:443] INACTIVE
22:56:58.690 [reactor-tcp-client-io-4] DEBUG reactor.io.netty.http.HttpClient - [id: 0xd90b01c3, L:/9.47.225.81:55186 ! R:api.ng.bluemix.net/169.54.245.68:443] UNREGISTERED
22:56:58.690 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8, L:/9.47.225.81:55184 ! R:api.ng.bluemix.net/169.54.245.68:443] INACTIVE
22:56:58.696 [reactor-tcp-client-io-2] DEBUG reactor.io.netty.http.HttpClient - [id: 0xabd28cb8, L:/9.47.225.81:55184 ! R:api.ng.bluemix.net/169.54.245.68:443] UNREGISTERED
22:56:58.707 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd, L:/9.47.225.81:55185 - R:api.ng.bluemix.net/169.54.245.68:443] CLOSE
22:56:58.708 [reactor-tcp-client-io-3] DEBUG r.i.netty.common.NettyChannelHandler - Pausing read due to lack of request
22:56:58.708 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd, L:/9.47.225.81:55185 ! R:api.ng.bluemix.net/169.54.245.68:443] INACTIVE
22:56:58.708 [reactor-tcp-client-io-3] DEBUG reactor.io.netty.http.HttpClient - [id: 0x69fdf2dd, L:/9.47.225.81:55185 ! R:api.ng.bluemix.net/169.54.245.68:443] UNREGISTERED
```

**Comment by twoseat** [comment]
Thanks for that @lalyos. The problem is that the client you're using (i.e. cf-java-client) is not authenticated (which is separate from your user authentication). You need to provide a `clientId()` and `clientSecret()` on the `SpringCloudFoundryClient.builder()`. It works for me because we set default values that work on standard installs such as pcfdev, but I imagine they won't for BlueMix. I'm afraid I don't know where you would create/find these under BlueMix, though I've seen mention of a Management Console? Let me know how that goes, and in the meantime I'll check what tasks we have coming up to improve our doc to spell this out..

**Comment by lastras** [comment]
Thank you but why does cf work apparently without needing to know clientId and clientSecret?

**Comment by nebhale** [comment]
After a deeper bit of investigation it appears that the underlying `reactor-netty` implementation is choking on the network handshake with Bluemix.  We'll need to enlist @smaldini's help to track it down.

**Comment by nebhale** [comment]
Resolved by 01dec2d1c406838c772503afee1f05b405d71107
